deploy/07_deploy_token_staking.ts

```diff
@@ -12,23 +15,65 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
   const VendingMachineNuCypher = await deployments.get("VendingMachineNuCypher")
   const KeepStake = await deployments.get("KeepStake")
 
-  const TokenStaking = await deployments.deploy("TokenStaking", {
-    from: deployer,
-    args: [
-      T.address,
-      KeepTokenStaking.address,
-      NuCypherStakingEscrow.address,
-      VendingMachineKeep.address,
-      VendingMachineNuCypher.address,
-      KeepStake.address,
-    ],
-    log: true,
-  })
+  const tokenStakingConstructorArgs = [
```

Comment:
`hardhat-deploy` already supports proxy deployments for Open Zeppelin (https://github.com/wighawag/hardhat-deploy#deploying-and-upgrading-proxies). Can't we use it instead of `"@openzeppelin/hardhat-upgrades"` plugin?

This would mean that we just need to add
```ts
    proxy: hre.network.name === "hardhat" ? false : {
      proxyContract: 'OpenZeppelinTransparentProxy'
    },
```
to `deployment.deploy` function options argument.

Reply:
`hardhat-deploy` merely deploys the proper contracts, but OpenZeppelin Upgrades goes further to make sure that the contract is upgrade safe and that the storage layout is compatible with existing state. This implies checking the storage arrangement of the previous implementation with respect to any proposed upgrade
(see https://github.com/OpenZeppelin/openzeppelin-upgrades#how-do-the-plugins-work). To do so, it keeps track of storage variables in a separate file in `.openzeppelin/mainnet.json`, which I include in this PR.